File: modules/crypto-core/src/main/java/com/shterneregen/securelan/crypto/service/RsaCryptoService.java
package com.shterneregen.securelan.crypto.service;

import java.security.PrivateKey;
import java.security.PublicKey;

public interface RsaCryptoService {
    byte[] encrypt(byte[] plainText, PublicKey publicKey);

    byte[] decrypt(byte[] cipherText, PrivateKey privateKey);
}

File: modules/crypto-core/src/main/kotlin/com/shterneregen/securelan/crypto/service/RsaCryptoService.kt
package com.shterneregen.securelan.crypto.service

import java.security.PrivateKey
import java.security.PublicKey

interface RsaCryptoService {
    fun encrypt(plainText: ByteArray, publicKey: PublicKey): ByteArray

    fun decrypt(cipherText: ByteArray, privateKey: PrivateKey): ByteArray
}
